src/api/test-controller.js

```diff
@@ -98,86 +106,70 @@ export default class TestController {
+    _doubleClick$ (selector, options) {
         return this._enqueueAction('doubleClick', DoubleClickCommand, { selector, options });
     }
 
-    _hoverImpl (selector, options) {
+    _hover$ (selector, options) {
         return this._enqueueAction('hover', HoverCommand, { selector, options });
     }
 
-    _dragImpl (selector, dragOffsetX, dragOffsetY, options) {
+    _drag$ (selector, dragOffsetX, dragOffsetY, options) {
         return this._enqueueAction('drag', DragCommand, { selector, dragOffsetX, dragOffsetY, options });
     }
 
-    _dragToElementImpl (selector, destinationSelector, options) {
+    _dragToElement$ (selector, destinationSelector, options) {
         return this._enqueueAction('dragToElement', DragToElementCommand, { selector, destinationSelector, options });
     }
 
-    _typeTextImpl (selector, text, options) {
+    _typeText$ (selector, text, options) {
         return this._enqueueAction('typeText', TypeTextCommand, { selector, text, options });
     }
 
-    _selectTextImpl (selector, startPos, endPos) {
+    _selectText$ (selector, startPos, endPos) {
         return this._enqueueAction('selectText', SelectTextCommand, { selector, startPos, endPos });
     }
 
-    _selectTextAreaContentImpl (selector, startLine, startPos, endLine, endPos) {
+    _selectTextAreaContent$ (selector, startLine, startPos, endLine, endPos) {
         return this._enqueueAction('selectTextAreaContent', SelectTextAreaContentCommand, { selector, startLine, startPos, endLine, endPos });
     }
 
-    _selectEditableContentImpl (startSelector, endSelector) {
+    _selectEditableContent$ (startSelector, endSelector) {
         return this._enqueueAction('selectEditableContent', SelectEditableContentCommand, { startSelector, endSelector });
     }
 
-    // API
-    click (selector, options) {
-        return this._clickImpl(selector, options);
-    }
-
-    rightClick (selector, options) {
-        return this._rightClickImpl(selector, options);
-    }
-
-    doubleClick (selector, options) {
-        return this._doubleClickImpl(selector, options);
-    }
-
-    hover (selector, options) {
-        return this._hoverImpl(selector, options);
-    }
-
-    drag (selector, dragOffsetX, dragOffsetY, options) {
-        return this._dragImpl(selector, dragOffsetX, dragOffsetY, options);
-    }
-
-    dragToElement (selector, destinationSelector, options) {
-        return this._dragToElementImpl(selector, destinationSelector, options);
-    }
+    _eval$ (fn, dependencies) {
+        var hybridFn = createHybridFunction(fn, dependencies, this.testRun, {
+            instantiation: 'eval',
+            execution:     'eval'
+        });
 
-    typeText (selector, text, options) {
-        return this._typeTextImpl(selector, text, options);
+        return hybridFn();
     }
+}
 
-    selectText (selector, startPos, endPos) {
-        return this._selectTextImpl(selector, startPos, endPos);
-    }
+(function createAPIMethods () {
```

Comment:
Nope. We add methods to `prototype`, if we do it constructor we will add unnecessary memory overhead